src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -41,65 +50,149 @@ export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
   };
 
   const guidedExperienceBreaches = getGuidedExperienceBreaches(
-    props.data.subscriberBreaches,
+    subscriberBreaches,
     props.subscriberEmails,
   );
 
+  // TODO: Write unit tests MNTOR-2560
+  /* c8 ignore start */
   const nextStep = getNextGuidedStep(props.data, stepMap[props.type]);
-  const pageData = getLeakedPasswords({
+  const [emailAffected, setEmailAffected] = useState<string>();
+  const [pageDataContent, setPageDataContent] = useState<LeakedPasswordLayout>({
     dataType: props.type,
     breaches: guidedExperienceBreaches,
     l10n: l10n,
-    nextStep,
+    emailAffected: emailAffected ?? "",
+    nextStep: nextStep,
   });
 
-  // The non-null assertion here should be safe since we already did this check
-  // in `./[type]/page.tsx`:
-  const { title, illustration, content } = pageData!;
+  const pageData = getLeakedPasswords(pageDataContent);
+  const unresolvedPasswordBreachContent =
+    pageData.unresolvedPasswordBreachContent;
+  const unresolvedPasswordBreach =
+    pageData.unresolvedPasswordBreach ??
+    pageData.unresolvedSecurityQuestionsBreach;
+
+  // Data class string to push to resolutionsChecked array
+  const resolvedDataClassName =
+    props.type === "passwords" ? "passwords" : "security-questions-and-answers";
+
   const isStepDone =
     props.type === "passwords-done" || props.type === "security-questions-done";
 
+  useEffect(() => {
+    if (unresolvedPasswordBreach) {
+      // Check for the email assosciated with the leaked password/security question breach
+      const emailMatch = props.subscriberEmails.find((email) =>
+        unresolvedPasswordBreach.emailsAffected.includes(email),
+      );
+
+      if (emailMatch) {
+        setEmailAffected(emailMatch);
+        const newPageData: LeakedPasswordLayout = {
+          dataType: props.type,
+          breaches: guidedExperienceBreaches,
+          l10n: l10n,
+          emailAffected: emailMatch,
+          nextStep: nextStep,
+        };
+        setPageDataContent(newPageData);
+      }
+    }
+    // This should only run whenever unresolvedPasswordBreach changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [unresolvedPasswordBreach]);
```

Comment:
Looking at this `useEffect` it’s still more complicated than it needs to be. We should be able to get rid of it entirely. The main reason I see this hook being needed here is that we are depend on `emailAffected` and `pageContend` being stored as state. We only need `subscriberBreaches` to be updated in order to derive the current view state.

This should look something like this:
```suggestion
  const unresolvedBreach = findFirstUnresolvedBreach(
    guidedExperienceBreaches,
    props.type,
  );
  // Check for the email assosciated with the leaked password/security question breach
  const emailAffected = props.subscriberEmails.find((email) =>
    unresolvedBreach?.emailsAffected.includes(email),
  ) ?? "";
  const unresolvedPasswordBreachContent = getLeakedPasswords({
    dataType: props.type,
    breaches: guidedExperienceBreaches,
    l10n: l10n,
    emailAffected,
    nextStep: nextStep,
  });
```